### PR TITLE
Updated documentation for `Toast` component

### PR DESCRIPTION
**Pull request by smmr-dn**
## Changes

Updated documentation for `Toast` component.

## Testing

N/A

## Docs

N/A

### CODE REVIEW

File: apps/website/src/content/docs/toast.mdx [hunk 1]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
 
-<Placeholder componentPageName='toasts--positive' />
-
 <LiveExample src='Toast.main.jsx'>
   <AllExamples.ToastMainExample client:load />
 </LiveExample>
 
 A toast notification provides a brief message about app processes at the top of the screen. Because these messages are so prominent, we need to be careful not to overuse or misuse them.
 
+## Variants
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have removed the "Variants" section and moved everything under "Usage".

File: apps/website/src/content/docs/toast.mdx [hunk 2]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
+
+<LiveExample src='Toast.negative.jsx'>
+  <AllExamples.ToastNegativeExample client:load />
+</LiveExample>
+
+<LiveExample src='Toast.warning.jsx'>
+  <AllExamples.ToastWarningExample client:load />
+</LiveExample>
+
+<LiveExample src='Toast.informational.jsx'>
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have combined them all in one `Live Example`.

File: apps/website/src/content/docs/toast.mdx [hunk 3]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
+
+`Start` indicates left side of viewport. `End` - right side of viewport.
+
+<LiveExample src='Toast.positionChanged.jsx'>
+  <AllExamples.ToastPositionChangedExample client:load />
+</LiveExample>
+
+### Anchor to button
+
+Use `animateOutTo` to set the direction to which the toast goes when closing.
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have described more about `animateOutTo`.

File: apps/website/src/content/docs/toast.mdx [hunk 4]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
+
+<LiveExample src='Toast.anchorToButton.jsx'>
+  <AllExamples.ToastAnchorToButtonExample client:load />
+</LiveExample>
+
+## Usage
+
+### Close individual
+
+Each toast can be closed individually and chained after one another to display multiple messages.
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have explained more about closing toasts.

File: apps/website/src/content/docs/toast.mdx [hunk 5]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
+
+Use `animateOutTo` to set the direction to which the toast goes when closing.
+
+<LiveExample src='Toast.anchorToButton.jsx'>
+  <AllExamples.ToastAnchorToButtonExample client:load />
+</LiveExample>
+
+## Usage
+
+### Close individual
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have combined those that are related to closing all under "Closing toasts".

File: apps/website/src/content/docs/toast.mdx [hunk 6]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
+
+<LiveExample src='Toast.informational.jsx'>
+  <AllExamples.ToastInformationalExample client:load />
+</LiveExample>
+
+### Position changed
+
+The toast position can be customized using the `placement` prop, which is default to `top`. The available `placement` values are:
+
+- `bottom-end`
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have made this change.

File: apps/website/src/content/docs/toast.mdx [hunk 7]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
-<Placeholder componentPageName='toasts--positive' />
-
 <LiveExample src='Toast.main.jsx'>
   <AllExamples.ToastMainExample client:load />
 </LiveExample>
 
 A toast notification provides a brief message about app processes at the top of the screen. Because these messages are so prominent, we need to be careful not to overuse or misuse them.
 
+## Variants
+
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have had a section for `link`.

File: apps/website/src/content/docs/toast.mdx [hunk 8]
```diff
@@ -6,14 +6,74 @@ thumbnail: Toast
...
+The toast position can be customized using the `placement` prop, which is default to `top`. The available `placement` values are:
+
+- `bottom-end`
+- `top`
+- `top-start`
+- `top-end`
+- `bottom`
+- `bottom-start`
+
+`Start` indicates left side of viewport. `End` - right side of viewport.
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have made the change.

File: apps/website/src/content/docs/toast.mdx [hunk 9]
```diff
@@ -1,18 +1,82 @@
...
+The toast position can be customized using the `placement` prop, which is default to `top`. The available `placement` values are:
+
+- `top`
+- `top-start`
+- `top-end`
+- `bottom`
+- `bottom-start`
+- `bottom-end`
+
+`start` indicates left side of viewport, while `end` represents the right side of viewport.
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have made the change.

**Review by smmr-dn — COMMENTED** [review]
As discussed in our documentation working session, I have made suggested changes to this PR:
- Combined sections that are related to status method `options` under `Status`. Reworded `Status` description to reflect more of the code, not just in plain English.
- Removed unnecessary code in examples and added more examples for `order` and `hasCloseButton`.
- Some more re-wording.

File: apps/website/src/content/docs/toast.mdx [hunk 10]
```diff
@@ -1,18 +1,100 @@
...
+  <AllExamples.ToastPlacementExample client:load />
+</LiveExample>
+
+`order` supports the following values:
+
+- `"auto"` (default)
+- `"descending"`
+- `"ascending"`
+
+When set to `"descending"`, the most recent toasts are on top. When set to `"ascending"`, the most recent toasts are on bottom. When set to `"auto"`, it will behave like `"descending"` when `placement` is set to a `-top` value, otherwise `"ascending"`.
```
[the remark this answers is not in the document]
Comment (mayank99) [reply apps/website/src/content/docs/toast.mdx:61]: It represents the "suffix" part of the values, since there is no value equal to `top` exactly.

File: apps/website/src/content/docs/toast.mdx [hunk 11]
```diff
@@ -1,18 +1,100 @@
...
+
+### Closing toasts
+
+When setting the `hasCloseButton` option available in the status method to `true`, each toast will be displayed with a close button, which allows you to close the toast manually.
+
+<LiveExample src='Toast.hasCloseButton.jsx'>
+  <AllExamples.ToastHasCloseButtonExample client:load />
+</LiveExample>
+
+Additionally, to implement further actions upon each toast closing or chain after one another to display multiple messages, you can retrieve the `close()` function returned from the status method.
```
[the remark this answers is not in the document]
Comment (mayank99) [reply apps/website/src/content/docs/toast.mdx:75]: I think there's a typo in your suggestion.

File: apps/website/src/content/docs/toast.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply apps/website/src/content/docs/toast.mdx:61]: It was supposed to represent the "suffix" part of the values, since `top` is not the only value. It seems like I may have gotten things confused though. It should be a prefix, not a suffix.

Even better suggestion, we could rephrase this as:

```suggestion
When set to `"descending"`, the most recent toasts are on top. When set to `"ascending"`, the most recent toasts are on bottom. When set to `"auto"`, it will behave like `"descending"` when `placement` is set to a value beginning with `"top"`, otherwise it will behave like `"ascending"`.
```

File: examples/Toast.hasCloseButton.jsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import * as React from 'react';
+import { useToaster, Button } from '@itwin/itwinui-react';
+
+export default () => {
+  const toaster = useToaster();
+
+  React.useEffect(() => {
+    toaster.setSettings({
+      hasCloseButton: true,
+    });
```
Comment (mayank99) [inline examples/Toast.hasCloseButton.jsx:14]: This does not look right.




Isn't `hasCloseButton` an individual toast-level option, rather than a toaster-level setting?

File: examples/Toast.close.jsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+          alignItems: 'flex-end',
+        }}
+      >
+        <ProgressRadial
+          size={'small'}
+          indeterminate
+          style={{ marginRight: '8px' }}
+        />
+        Your process is running...
+      </div>,
```
Comment (mayank99) [inline examples/Toast.close.jsx:26]: Could we just use flex here?

```jsx
<Flex>
  <ProgressRadial size='small'/>
  Your process is running…
</Flex>
```

File: examples/Toast.hasCloseButton.jsx (hunk 12 above)
Comment (smmr-dn) [reply examples/Toast.hasCloseButton.jsx:14]: Right, it should have been set in the status method. I have made the fix.

File: examples/Toast.close.jsx (hunk 13 above)
Comment (smmr-dn) [reply examples/Toast.close.jsx:26]: I have made the change.

File: apps/website/src/content/docs/toast.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx:61]: I have rephrased this as suggested.

File: apps/website/src/content/docs/toast.mdx [hunk 14]
```diff
@@ -1,18 +1,100 @@
...
+
+<LiveExample src='Toast.close.jsx'>
+  <AllExamples.ToastCloseExample client:load />
+</LiveExample>
+
+Closing also depends on which type of toast you are using, without the use of `hasCloseButton`. Persisting toasts will not be closed automatically, and will contain a close button. Temporary toasts will automatically close after 7 seconds and will not contain a close button. The type of toasts can be specified by passing either `"persisting"` or `"temporary"` into the `type` option.
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx:81]: I have done as suggested.

File: examples/Toast.close.jsx (hunk 13 above)
Comment (mayank99) [reply examples/Toast.close.jsx]: nit: The `ProgressRadial` can also be simplified as per my suggestion above

File: examples/Toast.placement.jsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import { useToaster, Button } from '@itwin/itwinui-react';
+
+export default () => {
+  const toaster = useToaster();
+
+  React.useEffect(() => {
+    toaster.setSettings({
+      placement: 'bottom-end',
```
Comment (mayank99) [inline examples/Toast.placement.jsx:13]: This only shows one `placement`. Maybe we should render a `<Select>` to showcase all placements?

File: examples/Toast.order.jsx [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import { useToaster, Button } from '@itwin/itwinui-react';
+
+export default () => {
+  const toaster = useToaster();
+
+  React.useEffect(() => {
+    toaster.setSettings({
+      order: 'ascending',
```
Comment (mayank99) [inline examples/Toast.order.jsx:13]: Similarly, we could showcase all `order` values.

File: apps/website/src/content/docs/toast.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have made the change.

File: examples/Toast.placement.jsx (hunk 15 above)
Comment (smmr-dn) [reply examples/Toast.placement.jsx]: I have added a `Select` for placements.

File: examples/Toast.order.jsx (hunk 16 above)
Comment (smmr-dn) [reply examples/Toast.order.jsx]: I have added a `Select` for orders.

File: apps/website/src/content/docs/toast.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have made the change.

File: examples/Toast.placement.jsx (hunk 15 above)
Comment (mayank99) [reply examples/Toast.placement.jsx]: Follow-up suggestion: You can remove the `useEffect` and the `selectedPlacement` state. Instead, you can directly call `toaster.setSettings` inside `onChange`. See https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers

Comment (smmr-dn) [reply examples/Toast.placement.jsx]: I have removed the useEffect.

File: apps/website/src/content/docs/toast.mdx [hunk 17]
```diff
@@ -1,18 +1,100 @@
...
+
+### Closing toasts
+
+When setting the `hasCloseButton` option available in the status method to `true`, each toast will be displayed with a close button, which allows you to close the toast manually.
+
+<LiveExample src='Toast.hasCloseButton.jsx'>
+  <AllExamples.ToastHasCloseButtonExample client:load />
+</LiveExample>
+
+Additionally, to implement further actions upon the closing of each toast or to chain one toast after another, you can retrieve the `close()` function returned from the status method.
```
Comment (mayank99) [inline apps/website/src/content/docs/toast.mdx:75]:
```suggestion
Additionally, to implement further actions upon the closing of each toast or to chain one toast after another, you can call the `close()` function returned from the status method.
```

File: apps/website/src/content/docs/toast.mdx [hunk 18]
```diff
@@ -1,18 +1,100 @@
...
+  <AllExamples.ToastTypeExample client:load />
+</LiveExample>
+
+To close all toasters on viewport at once, use the `closeAll()` method offered by the toaster object.
+
+<LiveExample src='Toast.closeAll.jsx'>
+  <AllExamples.ToastCloseAllExample client:load />
+</LiveExample>
+
+The `animateOutTo` prop allows you to control the direction to which the toast animates out when closing. Specifically, it helps determine an "anchor" element that the toast will go towards when it is dismissed by pointing to a ref passed into that element.
```
Comment (mayank99) [inline apps/website/src/content/docs/toast.mdx:93]: We can simplify wording and provide an example use-case of this feature.


```suggestion
The `animateOutTo` prop allows you to control the direction to which the toast animates out when closing. Specifically, it helps determine an "anchor" element that the toast will go towards when it is dismissed. This can be useful, for example, when you want to indicate to the user that the toasts are being saved into a "notifications" panel.
```

File: apps/website/src/content/docs/toast.mdx [hunk 19]
```diff
@@ -1,18 +1,100 @@
...
+  <AllExamples.ToastCloseExample client:load />
+</LiveExample>
+
+Closing also depends on the type of toast when used without `hasCloseButton`. By default, persisting toasts will not be closed automatically and will contain a close button while temporary toasts will automatically close after 7 seconds and will not contain a close button. The type of toasts can be specified by passing either `"persisting"` or `"temporary"` into the `type` option.
+
+<LiveExample src='Toast.type.jsx'>
+  <AllExamples.ToastTypeExample client:load />
+</LiveExample>
+
+To close all toasters on viewport at once, use the `closeAll()` method offered by the toaster object.
```
Comment (mayank99) [inline apps/website/src/content/docs/toast.mdx:87]:
```suggestion
To close all currently open toasts at once, use the `closeAll()` method offered by the toaster object.
```

File: apps/website/src/content/docs/toast.mdx [hunk 20]
```diff
@@ -1,18 +1,100 @@
...
+Before triggering a toast, the toaster needs to be initialized using the `useToaster` hook. This hook returns an object which provides various methods for triggering the toast and changing the toaster settings.
 
 <LiveExample src='Toast.main.jsx'>
   <AllExamples.ToastMainExample client:load />
 </LiveExample>
 
-A toast notification provides a brief message about app processes at the top of the screen. Because these messages are so prominent, we need to be careful not to overuse or misuse them.
+### Status
+
+Toast statuses can be set using the status methods provided by the toaster object. The status method takes in a toast message and an `options` object for further customization. Four available statuses are:
```
Comment (mayank99) [inline apps/website/src/content/docs/toast.mdx:19]: nit: We are not setting a toast status, we are triggering a toast.

```suggestion
Toasts can be triggered using the status methods provided by the toaster object. The status method takes in a toast message and an `options` object for further customization. Four available statuses are:
```

File: apps/website/src/content/docs/toast.mdx [hunk 21]
```diff
@@ -1,18 +1,100 @@
...
+description: A toast is a non-disruptive message that provides feedback based on a user action.
 thumbnail: Toast
 ---
 
 <p>{frontmatter.description}</p>
 
-<Placeholder componentPageName='toasts--positive' />
+## Usage
+
+Before triggering a toast, the toaster needs to be initialized using the `useToaster` hook. This hook returns an object which provides various methods for triggering the toast and changing the toaster settings.
```
Comment (mayank99) [inline apps/website/src/content/docs/toast.mdx:11]: Simplification + accuracy. (I realized that "initialized" is incorrect, because `useToaster` is simply grabbing a reference to the already-initialized toaster)

```suggestion
Before triggering a toast, you can need to call the `useToaster` hook, which returns an object which provides various methods for triggering the toast and changing the toaster settings.
```

File: apps/website/src/content/docs/toast.mdx [hunk 22]
```diff
@@ -1,18 +1,100 @@
...
 
 <LiveExample src='Toast.main.jsx'>
   <AllExamples.ToastMainExample client:load />
 </LiveExample>
```
Comment (mayank99) [inline apps/website/src/content/docs/toast.mdx:16]: Would be good to mention this caveat, since it is a common issue that our users run into.

```suggestion

**Note**: `useToaster` must be called from the top level of a component which is rendered a descendant of [`ThemeProvider`](/docs/themeprovider).

```

File: apps/website/src/content/docs/toast.mdx [hunk 23]
```diff
@@ -1,18 +1,100 @@
...
+
+When set to `"descending"`, the most recent toasts are on top. When set to `"ascending"`, the most recent toasts are on bottom. When set to `"auto"`, it will behave like `"descending"` when `placement` is set to a value beginning with `"top"`, otherwise it will behave like `"ascending"`.
+
+<LiveExample src='Toast.order.jsx'>
+  <AllExamples.ToastOrderExample client:load />
+</LiveExample>
+
+### Closing toasts
+
+When setting the `hasCloseButton` option available in the status method to `true`, each toast will be displayed with a close button, which allows you to close the toast manually.
```
Comment (mayank99) [inline apps/website/src/content/docs/toast.mdx:69]: Prefer "end user", because "you" typically refers to the developer who is using our components.

```suggestion
When setting the `hasCloseButton` option available in the status method to `true`, each toast will be displayed with a close button, which allows the end user to close the toast manually.
```

Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have all the wording/text changes.

File: apps/website/src/content/docs/toast.mdx [hunk 24]
```diff
@@ -1,18 +1,102 @@
...
+## Usage
+
+Before triggering a toast, you need to call the `useToaster` hook, which returns an object containing various methods for triggering the toast and changing the toaster settings.
 
 <LiveExample src='Toast.main.jsx'>
   <AllExamples.ToastMainExample client:load />
 </LiveExample>
 
-A toast notification provides a brief message about app processes at the top of the screen. Because these messages are so prominent, we need to be careful not to overuse or misuse them.
+**Note**: `useToaster` must be called from the top level of a component which is rendered a descendant of [`ThemeProvider`](/docs/themeprovider).
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/toast.mdx]: I have made the change :)
